conductr_cli/bndl_utils.py

```diff
@@ -75,74 +74,154 @@ def try_match(pattern, slice):
     elif b'/manifest.json\x00\x00\x00' in initial_chunk and b'/layer.tar\x00\x00\x00' in initial_chunk:
         # tar c on a docker folder (somewhat unreliable)
         return 'docker'
+    elif b'\x00' not in initial_chunk and initial_chunk != b'' and data_is_bundle_conf(initial_chunk):
+        return 'bundle'
+    elif data_is_zip(initial_chunk):
+        return 'bundle'
+    elif data_is_tar(initial_chunk):
+        # TAR marker
+        return 'bundle'
     else:
         return None
 
 
-def load_bundle_args_into_conf(config, args):
-    if args.name is not None:
-        config.put('name', args.name)
+def data_is_bundle_conf(data):
+    try:
+        ConfigFactory.parse_string(data.decode('UTF-8'))
+        return True
+    except:
+        return False
 
-    for argument, bundle_key in mappings.items():
-        value = getattr(args, bundle_key, None)
 
-        if value is not None:
-            config.put(bundle_key, value)
+def data_is_tar(data):
+    return len(data) >= 262 and data[257:262] == b'ustar'
```

Comment:
Magic number per https://en.wikipedia.org/wiki/Tar_(computing) . We don't need to be perfect with auto-detection, just need to work in most cases. There's always the `-f` flag if necessary.